File: BAEKJOON/src/bronze/i/py/BOJ_11050_BinomialCoefficient.py
# 이항 계수 1
import math
import sys

n, k = map(int, sys.stdin.readline().split())

# result = 1
# for i in range(min(k, n - k)):
#     result *= n - i
#     result //= i + 1

result = math.factorial(n) // (math.factorial(k) * math.factorial(n - k))

print(result)

File: BAEKJOON/src/bronze/i/java/BOJ_11050_BinomialCoefficient.java
package bronze.i.java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

// 이항 계수 1
public class BOJ_11050_BinomialCoefficient {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());
        int res = 1;
        K = Math.min(K, N - K);
        for (int i = 0; i < K; i++) {
            res *= N - i;
            res /= i + 1;
        }
        System.out.print(res);
    }
}
